src/gui/trackerlist/trackerlistwidget.cpp

```diff
@@ -0,0 +1,452 @@
+    connect(header(), &QHeaderView::sectionResized, this, &TrackerListWidget::saveSettings);
+    connect(header(), &QHeaderView::sortIndicatorChanged, this, &TrackerListWidget::saveSettings);
+
+    // Set hotkeys
+    const auto *editHotkey = new QShortcut(Qt::Key_F2, this, nullptr, nullptr, Qt::WidgetShortcut);
+    connect(editHotkey, &QShortcut::activated, this, &TrackerListWidget::editSelectedTracker);
+    const auto *deleteHotkey = new QShortcut(QKeySequence::Delete, this, nullptr, nullptr, Qt::WidgetShortcut);
+    connect(deleteHotkey, &QShortcut::activated, this, &TrackerListWidget::deleteSelectedTrackers);
+    const auto *copyHotkey = new QShortcut(QKeySequence::Copy, this, nullptr, nullptr, Qt::WidgetShortcut);
+    connect(copyHotkey, &QShortcut::activated, this, &TrackerListWidget::copyTrackerUrl);
+
+    connect(this, &QAbstractItemView::doubleClicked, this, &TrackerListWidget::editSelectedTracker);
+}
+
+TrackerListWidget::~TrackerListWidget()
+{
+    saveSettings();
+}
+
+void TrackerListWidget::setTorrent(BitTorrent::Torrent *torrent)
+{
+    m_model->setTorrent(torrent);
+}
+
+BitTorrent::Torrent *TrackerListWidget::torrent() const
+{
+    return m_model->torrent();
+}
+
+QModelIndexList TrackerListWidget::getSelectedTrackerRows() const
+{
+    QModelIndexList selectedItemIndexes = selectionModel()->selectedRows();
+    selectedItemIndexes.removeIf([](const QModelIndex &index)
+    {
+        return (index.parent().isValid() || (index.row() < TrackerListModel::STICKY_ROW_COUNT));
+    });
+
+    return selectedItemIndexes;
+}
+
+void TrackerListWidget::decreaseSelectedTrackerTiers()
+{
+    const auto &trackerIndexes = getSelectedTrackerRows();
+    if (trackerIndexes.isEmpty())
+        return;
+
+    QSet<QString> trackerURLs;
+    for (const QModelIndex &index : trackerIndexes)
+    {
+        trackerURLs.insert(index.siblingAtColumn(TrackerListModel::COL_URL).data().toString());
+    }
+
+    QList<BitTorrent::TrackerEntry> trackers = m_model->torrent()->trackers();
+    for (BitTorrent::TrackerEntry &trackerEntry : trackers)
+    {
+        if (trackerURLs.contains(trackerEntry.url))
+        {
+            if (trackerEntry.tier > 0)
+                --trackerEntry.tier;
+        }
+    }
+
+    m_model->torrent()->replaceTrackers(trackers);
+}
+
+void TrackerListWidget::increaseSelectedTrackerTiers()
+{
+    const auto &trackerIndexes = getSelectedTrackerRows();
+    if (trackerIndexes.isEmpty())
+        return;
+
+    QSet<QString> trackerURLs;
+    for (const QModelIndex &index : trackerIndexes)
+    {
+        trackerURLs.insert(index.siblingAtColumn(TrackerListModel::COL_URL).data().toString());
+    }
+
+    QList<BitTorrent::TrackerEntry> trackers = m_model->torrent()->trackers();
+    for (BitTorrent::TrackerEntry &trackerEntry : trackers)
+    {
+        if (trackerURLs.contains(trackerEntry.url))
+        {
+            if (trackerEntry.tier < std::numeric_limits<decltype(trackerEntry.tier)>::max())
```

Comment:
> Overflows because of data type https://github.com/arvidn/libtorrent/blob/5ce4c516479085eda9327d35891b7732d6d5037b/include/libtorrent/aux_/announce_entry.hpp#L190
> 
> https://github.com/qbittorrent/qBittorrent/blob/f3b688217eb518ca67de23edd716278efde0b990/src/base/bittorrent/trackerentry.h#L72
> 
> But it's `int` somewhere else so... libtorrent problem? https://github.com/arvidn/libtorrent/blob/5ce4c516479085eda9327d35891b7732d6d5037b/include/libtorrent/add_torrent_params.hpp#L157

Good catch 👍 
At least it should be reported to `libtorrent`.

Reply:
>At least it should be reported to libtorrent.https://github.com/arvidn/libtorrent/issues/7523#issue-1909151754